jwst/cube_build/ifu_cube.py

```diff
@@ -2117,20 +2143,19 @@ def setup_final_ifucube_model(self, model_ref):
                 log.info('Number of wavelength planes removed with no data: %i',
                          remove_total)
 
-                flux = flux[remove_start:self.naxis3 - remove_final, :, :]
-                wmap = wmap[remove_start:self.naxis3 - remove_final, :, :]
-                dq = dq[remove_start:self.naxis3 - remove_final, :, :]
-                var = var[remove_start:self.naxis3 - remove_final, :, :]
-
-                # update WCS information if removing wavelengths from the IFU Cube
-                self.naxis3 = self.naxis3 - (remove_start + remove_final) + 1
+                flux = flux[remove_start: (self.naxis3 - remove_final), :, :]
+                wmap = wmap[remove_start: (self.naxis3 - remove_final), :, :]
+                dq = dq[remove_start: (self.naxis3 - remove_final), :, :]
+                var = var[remove_start: (self.naxis3 - remove_final), :, :]
 
                 if self.linear_wavelength:
                     self.crval3 = self.zcoord[remove_start]
                 else:
-                    self.wavelength_table = self.wavelength_table[remove_start:self.naxis3 - remove_final]
+                    self.wavelength_table = self.wavelength_table[remove_start: (self.naxis3 - remove_final)]
                     self.crval3 = self.wavelength_table[0]
 
+                # update WCS information if removing wavelengths from the IFU Cube
+                self.naxis3 = self.naxis3 - (remove_start + remove_final)
```

Comment:
Is this the same fix that was recently implemented in another PR? And if so, is the arithmetic correct here in terms of needing to add or subtract 1 to match up with the python slicing behavior?

Reply:
The is correct. No need to add 1.  I was wrong before

Reply:
So for all the data I have tested a cube is created - it is just really small. The extract1d step does run on the small cube. Let me force a situation where all the data is rejected and no cube is created and see what happens in extract1d